### PR TITLE
Cbm removal failure

### PR DESCRIPTION
**Pull request by Fris0uman**
#### Summary
```SUMMARY: Balance "Failure of removal of a CBM uses same logic for damage as failure of install"```

#### Purpose of change

Damage caused by a failed removal is not based on the difficulty of the removal while damaged caused by failed install is.

#### Describe the solution

Uses same logic as in bionics_install_failure() in bionics_uninstall_failure(). With the difference that level 5 failure casuse damge between 30 and 80 instead of installing a faulty CBM and PROF_MED and PROF_AUTODOC don't affect the outcome.
#### Additional context

This change is a copy-paste, both install_failure and uninstall_failure are now very similar, I don't know if they should be merged or stay as two similar but separate function.

### CODE REVIEW

**Comment by Photoloss** [comment]
Why do the profession perks not help? Surely a trained surgeon would have more experience in removing foreign objects and since the autodoc *can* perform the removal it's reasonable to assume similar procedures were part of the autodoc operator training.

Of course the benefit might be a lot smaller than for installation since under normal conditions you'd largely rely on patient-specific diagnostics to remove the faulty bionic as opposed to the myriad technical specs a layperson might mess up when installing a CBM without RTFM.

File: src/bionics.cpp [hunk 1]
```diff
@@ -797,8 +797,24 @@ void player::process_bionic( int b )
...
 {
+    // "success" should be passed in as a negative integer representing how far off we
+   // were for a successful removal.  We use this to determine consequences for failing.
+   success = abs( success );
+
+   // failure level is decided by how far off the character was from a successful removal, and
+   // this is scaled up or down by the ratio of difficulty/skill.  At high skill levels (or low
+   // difficulties), only minor consequences occur.  At low skill levels, severe consequences
+   // are more likely.
+   int failure_level = static_cast<int>( sqrt( success * 4.0 * difficulty / adjusted_skill ) );
```
Comment (Night-Pryanik) [inline src/bionics.cpp]:
```suggestion
   const int failure_level = static_cast<int>( sqrt( success * 4.0 * difficulty / adjusted_skill ) );
```

File: src/bionics.cpp [hunk 2]
```diff
@@ -797,8 +797,24 @@ void player::process_bionic( int b )
...
+   int fail_type = ( failure_level > 5 ? 5 : failure_level );
```
Comment (Night-Pryanik) [inline src/bionics.cpp]:
```suggestion
   const int fail_type = std::min( 5, failure_level );
```

File: src/bionics.cpp [hunk 3]
```diff
@@ -822,8 +838,29 @@ void player::bionics_uninstall_failure( player &installer )
                                              _( "<npcname> screws up the removal." ) );
             break;
     }
-    add_msg( m_bad, _( "%s body is severely damaged!" ), disp_name( true ) );
-    hurtall( rng( 30, 80 ), this ); // stop hurting yourself!
+
+    switch( fail_type ) {
+        case 1:
+            if( !( has_trait( trait_id( "NOPAIN" ) ) ) ) {
```
Comment (Night-Pryanik) [inline src/bionics.cpp]:
```suggestion
            if( !has_trait( trait_id( "NOPAIN" ) ) ) {
```

**Comment by Fris0uman** [comment]
>Why do the profession perks not help? 

In installation those perk downgrade faulty installation to damage. But there is no faulty installation here, of course they could still downgrade level 5 failure to level 4 if that seem reasonnable enough.

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/malfunctionning-cbms-removal/15358/5

**Pull request merged**